CodeMaid/Logic/Cleaning/UsingStatementCleanupLogic.cs

```diff
@@ -91,20 +85,25 @@ public void RemoveAndSortUsingStatements(TextDocument textDocument)
             else
             {
                 _commandHelper.ExecuteCommand(textDocument, "Edit.RemoveUnusedUsings");
-                _commandHelper.ExecuteCommand(textDocument, "Edit.SortUsings");
             }
 
-            // Check each using statement point and re-insert it if removed.
-            foreach (var point in points)
+            // Ignore any using statements that are still referenced
+            usingStatementsToReinsert = usingStatementsToReinsert
+                 .Where(usingStatement => TextDocumentHelper.FirstOrDefaultMatch(textDocument, string.Format(patternFormat, usingStatement)) == null)
+                 .ToList();
+
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
```suggestion
```
Although the IDE provides warnings about this, in general I've found we can invoke on background threads and boost performance.

Reply:
Ah, that makes sense. I'll remove it.